internal/fwserver/server_planresourcechange.go

```diff
@@ -209,6 +231,7 @@ func (s *Server) PlanResourceChange(ctx context.Context, req *PlanResourceChange
 			Diagnostics:     resp.Diagnostics,
 			Plan:            modifyPlanReq.Plan,
 			RequiresReplace: path.Paths{},
+			Private:         privateProviderData,
```

Comment:
I've found that setting the request up correctly then copying the request field into the response makes the logic in these a little easier to follow

```suggestion
			Private:         modifyPlanReq.Private,
```